File: objective-c/WesleySafadao.m
/* 
 * Author: Luiz Peres
 * And yes, I was listening to the song. 
 */

#import "WesleySafadao.h"

@implementation WesleySafadao

+ (int) sum:(int)n
{
  int res = 0;
  
  for (int i = 1; i <= n; i++)
    res += i;
  
  return res;
}

+ (WesleyType) getSafadezaWithDay:(int)day Month:(int)month AndYear:(int)year
{
  WesleyType safadao;
  
  safadao.safadeza = [WesleySafadao sum:month] + ((float)year / 100) * (50 - day);
  NSString *strSafado = [NSString stringWithFormat:@"%f", safadao.safadeza];
  NSLog(@"%@", strSafado);
  safadao.angel = 100.0 - safadao.safadeza;
  
  return safadao;
}

@end

File: javascript/es2015/wesley-safadao.js
'use strict';

let safadeza = (day, month, year) => {
  const sum = Array.apply(null, {
    length: month
  }).map((_, i) => i + 1).reduce((x, y) => x + y);
  return sum + ((year % 100) / 100) * (50 - day);
};

let angel = (day, month, year) => 100 - safadeza(day, month, year);

export angel;
export safadeza;
